client/src/main/java/io/pravega/schemaregistry/client/SchemaRegistryClientImpl.java

```diff
@@ -70,14 +82,25 @@
             .withExpBackoff(100, 2, 10, 1000)
             .retryWhen(x -> Exceptions.unwrap(x) instanceof ConnectionException);
     private static final int GROUP_LIMIT = 100;
+    private static final String HTTPS = "https";
+    private static final String TLS = "TLS";
 
     private final ApiV1.GroupsApi groupProxy;
     private final ApiV1.SchemasApi schemaProxy;
     private final String namespace;
     private final Client client;
     
     SchemaRegistryClientImpl(SchemaRegistryClientConfig config, String namespace) {
-        client = ClientBuilder.newClient(new ClientConfig());
+        Preconditions.checkNotNull(config);
+        Preconditions.checkNotNull(config.getSchemaRegistryUri());
+        ClientBuilder clientBuilder = ClientBuilder.newBuilder().withConfig(new ClientConfig());
+        if (HTTPS.equalsIgnoreCase(config.getSchemaRegistryUri().getScheme())) {
+            clientBuilder = clientBuilder.sslContext(getSSLContext(config));
+            if (!config.isValidateHostName()) {
```

Comment:
Here we are disabling hostname verification is `isValidateHostName` is `false`. Looking at the code here gives the impression that we don't do anything if it is `true`. Is there a way we could collate the off and on logic in one place?

Reply:
the default behaviour of HTTPS SSL context is to validate the host name. 
So in case where users do not wish to validate the host name, we use a verifier that always returns true.

Reply:
i have added the DefaultHostNameVerifier from pravega.keycloak which is basically a shading of apache default hostname verifier.

Reply:
Just for the record, the latest commit by Shivesh removes it as hostname verification is on by default.